onadata/libs/utils/logger_tools.py

```diff
@@ -275,6 +280,37 @@ def check_submission_permissions(request, xform):
               }))
 
 
+def check_submission_encryption(xform: XForm, xml: bytes) -> NoReturn:
+    """
+    Check that the submission is encrypted or unencrypted depending on the
+    encryption status of an XForm.
+
+    The submission is invalid if the XForm's encryption status is different
+    from the submissions
+    """
+    encryption_status = False
+    submission_element = ET.fromstring(xml)
+    encrypted_attrib = submission_element.attrib.get('encrypted')
+    required_encryption_elems = [
+        elem.tag for elem in submission_element
+        if elem.tag in REQUIRED_ENCRYPTED_FILE_ELEMENTS]
+
+    # Check the validity of the submission
+    if encrypted_attrib == "yes" or len(required_encryption_elems) > 1:
+        if (not len(required_encryption_elems) == 2 or
+                not encrypted_attrib == "yes") and xform.encrypted:
+            raise InstanceFormatError(
+                "Encrypted submission incorrectly formatted.")
+        encryption_status = True
+
+    if encryption_status != xform.encrypted:
+        form_status = "encrypted" if xform.encrypted else "unencrypted"
+        submisison_status = "Encrypted" if encryption_status else "Unencrypted"
```

Comment:
Since this is part of output to a user, you probably want to use the translated values.

Again, the only situation that matters is the "encrypted" form, if the form is not encrypted there is no need to reject the submission on this basis.

Reply:
> Again, the only situation that matters is the "encrypted" form, if the form is not encrypted there is no need to reject the submission on this basis.

How about in a situation where an encrypted submission is made to an unencrypted form? _Doubt it would happen in a normal submission flow but is it something we should reject?_

Reply:
No, not really. It is not going to affect exports adversely. That element does bring to question what other validations we need to do.